src/sagemaker/serializers.py

```diff
@@ -38,3 +41,57 @@ def serialize(self, data):
     @abc.abstractmethod
     def CONTENT_TYPE(self):
         """The MIME type of the data sent to the inference endpoint."""
+
+
+class NumpySerializer(BaseSerializer):
+    """Serialize data to a buffer using the .npy format."""
+
+    CONTENT_TYPE = "application/x-npy"
+
+    def __init__(self, dtype=None):
+        """Initialize the dtype.
+
+        Args:
+            dtype (str): The dtype of the data.
+        """
+        self.dtype = dtype
+
+    def serialize(self, data):
+        """Serialize data to a buffer using the .npy format.
+
+        Args:
+            data (object): Data to be serialized. Can be a NumPy array, list,
+                file, or buffer.
+
+        Returns:
+            _io.BytesIO: A buffer containing data serialzied in the .npy format.
+        """
+        if isinstance(data, np.ndarray):
+            if not data.size > 0:
+                raise ValueError("Cannot serialize empty array.")
+            return self.serialize_array(data)
+
+        if isinstance(data, list):
+            if not len(data) > 0:
+                raise ValueError("empty array can't be serialized")
```

Comment:
nit: I'd match this error message to the one above

Reply:
Fixed